bolt04/enctlvs.json

```diff
@@ -0,0 +1,61 @@
+    "blinding_secret": "76d4de6c329c79623842dcf8f8eaee90c9742df1b5231f5350df4a231d16ebcf01",
+    "blinding": "03fc5e56da97b462744c9a6b0ba9d5b3ffbfb1a08367af9cc6ea5ae03c79a78eec",
+    "encmsg": {
+      "next_node_id": "027f31ebc5462c1fdce1b737ecff52d37d75dea43ce11c74d25aa297165faa2007",
+      "blinding": "070707070707070707070707070707070707070707070707070707070707070701"
+    },
+    "ss": "f18a1ddb1cb27d8fc4faf2cf317e87524fcc6b7f053496d95bf6e6809d09851e",
+    "HMAC256('blinded_node_id', ss)": "8074773a3745818b0d97dd875023486cc35e7afd95f5e9ec1363f517979e8373",
+    "blinded_node_id": "026ea8e36f78e038c659beba9229699796127471d9c7a24a0308533371fd63ad48",
+    "E": "03fc5e56da97b462744c9a6b0ba9d5b3ffbfb1a08367af9cc6ea5ae03c79a78eec",
+    "H(E || ss)": "9afb8b2ebc174dcf9e270be24771da7796542398d29d4ff6a4e7b6b4b9205cfe",
+    "next_e": "5149cb0ce6100209f9a2a1140c4e6e8235381aac6201da2c0f8ce1f91bcc3b6e01",
+    "encmsg_hex": "0421027f31ebc5462c1fdce1b737ecff52d37d75dea43ce11c74d25aa297165faa20070c2102989c0b76cb563971fdc9bef31ec06c3560f3249d6ee9e5d83c57625596e05f6f",
+    "rho": "46b7af481999101f3d7fb20a87c4d6fb9191b93ddc0f163d00b94fcbc571f564",
+    "enctlv_hex": "1630da85e8759b8f3b94d74a539c6f0d870a87cf03d4986175865a2985553c997b560c36613bd9184c1a6d41a37027aabdab5433009d8409a1b638eb90373778a05716af2c215b3d31db7b2c2659716e663ba3d9c909"
+  },
+  {
+    "test name": "Padded enctlv for Carol, next is Dave",
+    "node_privkey": "434343434343434343434343434343434343434343434343434343434343434301",
+    "node_id": "027f31ebc5462c1fdce1b737ecff52d37d75dea43ce11c74d25aa297165faa2007",
+    "blinding_secret": "070707070707070707070707070707070707070707070707070707070707070701",
+    "blinding": "02989c0b76cb563971fdc9bef31ec06c3560f3249d6ee9e5d83c57625596e05f6f",
+    "encmsg": {
+      "next_node_id": "032c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991",
+      "padding": "0000000000000000000000000000000000000000000000000000000000000000000000"
+    },
+    "ss": "8c0f7716da996c4913d720dbf691b559a4945bf70cdd18e0b61e3e42635efc9c",
+    "HMAC256('blinded_node_id', ss)": "02afb2187075c8af51488242194b44c02624785ccd6fd43b5796c68f3025bf88",
+    "blinded_node_id": "02f4f524562868a09d5f54fb956ade3fa51ef071d64d923e395cc6db5e290ec67b",
+    "E": "02989c0b76cb563971fdc9bef31ec06c3560f3249d6ee9e5d83c57625596e05f6f",
+    "H(E || ss)": "cc3b918cda6b1b049bdbe469c4dd952935e7c1518dd9c7ed0cd2cd5bc2742b82",
+    "next_e": "306f017c285aac7b613e0f70895b7e4fa5cce9ddaf96f50c613d5dcee847703801",
+    "encmsg_hex": "012300000000000000000000000000000000000000000000000000000000000000000000000421032c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991",
+    "rho": "d6297b5645c27be8ea1240cc96e162c53fd9c389c6bec25cbdb4b6b3f654e7bc",
+    "enctlv_hex": "8285acbceb37dfb38b877a888900539be656233cd74a55c55344fb068f9d8da365340d21db96fb41b76123207daeafdfb1f571e3fea07a22e10da35f03109a0380b3c69fcbed9c698086671809658761cf65ecbc3c07a2e5"
+  }
+]
```

Comment:
It would be useful to have the whole onion combining these 3 payloads.

Reply:
Yes, agreed.

Reply:
I agree, I think most of these test vectors are just redundant with the test vectors in https://github.com/lightning/bolts/pull/765, onion messages should instead add test vectors for the onion-message-specific bits (a complete `onion_message` with an onion in it, and keys to decrypt it at intermediate nodes and at the recipient).

Reply:
Yes, with those test vectors I think this should be reduced to the same form (ideally, same data) as those with onionmsg specifics.  I'll do that tomorrow...